deeplake/auto/unstructured/coco/coco.py

```diff
@@ -0,0 +1,221 @@
+            )
+
+    def _add_annotation_tensors(
+        self,
+        structure: DatasetStructure,
+        inspect_limit: int = 1000000,
+    ):
+        for ann_file in self.annotation_files:
+            coco_file = CocoAnnotation(file_path=ann_file, creds=self.creds)
+            annotations = coco_file.annotations
+            file_name = Path(ann_file).stem
+            keys_in_group = set(chain.from_iterable(annotations[:inspect_limit]))
+
+            group_name = self.file_to_group.get(file_name, file_name)
+            group = GroupStructure(group_name)
+
+            for key in keys_in_group:
+                if key in self.ignore_keys:
+                    continue
+                tensor_name = self.key_to_tensor.get(key, key)
+
+                tensor = TensorStructure(
+                    name=tensor_name,
+                    params=DEFAULT_COCO_TENSOR_PARAMS.get(
+                        key, DEFAULT_GENERIC_TENSOR_PARAMS
+                    ),
+                )
+                group.add_item(tensor)
+
+            structure.add_group(group)
+
+    def _add_images_tensor(self, structure: DatasetStructure):
+        img_config = DEFAULT_IMAGE_TENSOR_PARAMS.copy()
+
+        if self.image_settings.get("linked", False):
+            img_config["htype"] = "link[image]"
+
+        img_config["sample_compression"] = self.image_settings.get(
+            "sample_compression", self.images.most_frequent_extension
+        )
+        name = self.image_settings.get("name", "images")
+
+        structure.add_first_level_tensor(
+            TensorStructure(name=name, primary=True, params=img_config)
+        )
+
+    def _ingest_images(
+        self,
+        ds: Dataset,
+        images: List[str],
+        progressbar: bool = True,
+        num_workers: int = 0,
+    ):
+        images_tensor_name = self.image_settings.get("name")
+        images_tensor = ds[images_tensor_name]
+        samples = OrderedDict((image, None) for image in images)
+
+        @deeplake.compute
+        def append_images(image, _, samples):
+            samples[image] = self.images.get_image(
+                image,
+                destination_tensor=images_tensor,
+                creds_key=self.image_settings.get("creds_key"),
+            )
+
+        append_images(samples).eval(
+            images, ds, progressbar=progressbar, num_workers=num_workers
+        )
+        images_tensor.extend(list(samples.values()), progressbar=progressbar)
+
+    def _get_sample(
+        self,
+        image_file: str,
+        ds: Dataset,
+        coco_file: CocoAnnotation,
+    ):
+        id_to_label = coco_file.id_to_label_mapping
+        image_name_to_id = coco_file.image_name_to_id_mapping
+        image_id = image_name_to_id[image_file]
+        matching_annotations = coco_file.get_annotations_for_image(image_id)
+        group_tensors = [
+            t for t in ds.tensors.keys() if not t.startswith("_")
```

Comment:
Use `tensor.meta.hidden` property to check if a tensor is hidden. User might create tensors leading with underscores.

Reply:
Changed. But, for some reason, I receive an exception inside transform.
``deeplake.util.exceptions.TransformError: '__tempbase_annotations/category_id_8799'``. Group name is ``base_annotations`` and tensor is ``category_id``.

Reply:
Fixed, after discussion with Fayaz